modules/ossm-preparing-to-update-istio-control-plane-with-inplace.adoc

```diff
@@ -0,0 +1,90 @@
+// Module included in the following assemblies:
+// update/ossm-updating-openshift-service-mesh.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="preparing-to-update-istio-control-plane-with-inplace_{context}"]
+= Preparing to update Istio control plane with InPlace strategy
+
+You can configure the {istio} control plane and workloads for the `InPlace` update strategy. When using the `InPlace` strategy, the `IstioRevision` resource created by the {SMProduct} Operator always uses the same name as the `{istio}` resource.
+
+.Procedure
+
+. Attach the workloads to a control plane deployed using the `InPlace` strategy:
+
+.. Label the namespace to automatically include all workloads by entering the following command:
++
+[source,terminal]
+----
+$ oc label namespace <namespace_name> istio.io/rev=<revision_name>
+----
+
+..  Apply the revision label to individual workloads by modifying the pod template in the `Deployment` resource. For example:
++
+[source,yaml, subs="attributes,verbatim"]
+----
+apiVersion: apps/v1
+kind: Deployment
+spec:
+  template:
+    metadata:
+      labels:
+        istio.io/rev: <revision_name>
+----
+
+. If the revision name is `default`, attach the workloads to the revision by running the following command. The following example labels the namespace with `istio-injection: enabled` label.
++
+[source,terminal]
+----
+$ oc label namespace <namespace_name> istio-injection=enabled
+----
+
+. Deploy the {istio} control plane using the `InPlace` update strategy. The following example configuration creates an `{istio}` resource named `default` in the `istio-system` namespace:
++
+[source,yaml, subs="attributes,verbatim"]
+----
+apiVersion: sailoperator.io/v1
+kind: Istio
+metadata:
+  name: default
+spec:
+  namespace: istio-system
+  version: v1.24.3
+  updateStrategy:
+    type: InPlace
+----
+
+. Install the {istio} CNI plugin with the desired version. The following example configuration creates an `IstioCNI` resource named `default` in the `istio-cni` namespace:
++
+[source,yaml, subs="attributes,verbatim"]
+----
+apiVersion: sailoperator.io/v1
+kind: IstioCNI
+metadata:
+  name: default
+spec:
+  version: v1.24.3
+  namespace: istio-cni
+----
+
+. Configure application workloads to run in the cluster. The follwing example deploys the `bookinfo` application in the `bookinfo` namespace.
+
+.. Create the `bookinfo` namespace by running the followin command:
++
+[source,terminal]
+----
+$ oc create ns bookinfo
+----
+
+.. Install the `bookinfo` pods in the `bookinfo` namespace by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f https://raw.githubusercontent.com/openshift-service-mesh/istio/release-1.24/samples/bookinfo/platform/kube/bookinfo.yaml -n bookinfo
+----
+
+. Label the `bookinfo` namespace to enable sidecar injection by running the following command:
```

Comment:
Should this be step 5c, instead of step 6? (Is it part of deploying the bookinfo application in the bookinfo namespace?)

Reply:
I am going to let it be as Step 6, as it has an individual line in the pre-requisites section in multiple modules. This module is just a prep to the procedure which was actually a pre reqs section.